### PR TITLE
IPF: convert uses of lexical_cast_default to from_chars

### PR DESCRIPTION
**Pull request by Vultraz**
CI run

### CODE REVIEW

File: src/image_modifications.cpp [hunk 1]
```diff
@@ -618,7 +617,7 @@ REGISTER_MOD_PARSER(TC, args)
 		return nullptr;
 	}
 
-	const int side_n = lexical_cast_default<int>(params[0], -1);
+	const int side_n = utils::from_chars<int>(params[0]).value_or(-1);
```
Comment (CelticMinstrel) [inline src/image_modifications.cpp:620]: This doesn't seem to be your fault, but… the error message 2 lines down is wrong, maybe you could fix that while you're at it? It should be outputting `params[0]`, not `side_n`.

Comment (Vultraz) [reply src/image_modifications.cpp:620]: Wil do.

**Comment by stevecotton** [comment]
It seems this raised the required version of GCC from GCC 7 to either 8 or 11. Would it be worth postponing the change until 1.19?

**Comment by Pentarctagon** [comment]
It's pretty late to be bumping compiler requirements, especially potentially from GCC 7 to GCC 11. I'd say revert -> I'll split 1.18 into its own branch with the release tomorrow -> revert the revert.

**Comment by CelticMinstrel** [comment]
I was recently thinking that this and Vultraz's previous IPF pull request both never should've been merged until 1.19, as they're a large refactor with great potential for creating new bugs. I recommend rolling them _both_ back and resurrecting them after 1.19 is branched off.

**Comment by Pentarctagon** [comment]
That'd be this one and #8396?

**Comment by CelticMinstrel** [comment]
Yeah. That one also makes use of `<charconv>`, right? So especially if that's the main issue, it needs to be covered too.

**Pull request merged**